WatchApp Extension/ComplicationController.swift

```diff
@@ -67,28 +103,107 @@ final class ComplicationController: NSObject, CLKComplicationDataSource {
     }
     
     func getTimelineEntries(for complication: CLKComplication, after date: Date, limit: Int, withHandler handler: (@escaping ([CLKComplicationTimelineEntry]?) -> Void)) {
-        // Call the handler with the timeline entries after to the given date
-        let entries: [CLKComplicationTimelineEntry]?
-
-        if  let context = ExtensionDelegate.shared().loopManager.activeContext,
-            let glucoseDate = context.glucoseDate
-        {
-            if glucoseDate.timeIntervalSince(date) > 0,
-                let template = CLKComplicationTemplate.templateForFamily(complication.family, from: context)
+        updateChartManagerIfNeeded {
+            let entries: [CLKComplicationTimelineEntry]?
+
+            if  let context = ExtensionDelegate.shared().loopManager.activeContext,
+                let glucoseDate = context.glucoseDate,
+                glucoseDate.timeIntervalSince(date) > 0,
+                let template = CLKComplicationTemplate.templateForFamily(complication.family, from: context, chartGenerator: self.makeChart)
             {
                 template.tintColor = UIColor.tintColor
                 entries = [CLKComplicationTimelineEntry(date: glucoseDate, complicationTemplate: template)]
             } else {
-                entries = []
+                entries = nil
             }
 
-            if let image = CLKComplicationTemplate.imageTemplate(for: complication.family) {
-                entries?.append(CLKComplicationTimelineEntry(date: glucoseDate.addingTimeInterval(.hours(1)), complicationTemplate: image))
-            }
-        } else {
-            entries = nil
+            handler(entries)
         }
+    }
+
+    // MARK: - Placeholder Templates
 
-        handler(entries)
+    func getLocalizableSampleTemplate(for complication: CLKComplication, withHandler handler: @escaping (CLKComplicationTemplate?) -> Void) {
```

Comment:
## Complication Placeholders
The header comment for `getLocalizableSampleTemplate(for family:)`:

> These methods will be called once per supported complication when your extension is installed, and the results cached.

Using `CLKRelativeDateTextProvider` in this method means the values get to be crazy, e.g. "-25 days".
You could use a simple text provider to fake a date string, or just not bother with this and let the Apple Watch app render a default (using app name/icon).